### PR TITLE
Firtst step at librekey

### PR DESCRIPTION
**Pull request by catap**
It is first step for #308 and G10 format support. In implementation of G10 format I have two missed parts:
 - `protected-private-key`;
 - 'advanced' format.

Also G10 right now doesn't support metainformation, but we can extend this format to support it — gnupg/g10 will ignore our field ;)

and yes, I moved keystore as `librekey`. I tried extract library for work with pgp, but it is a huge refactoring :(

### CODE REVIEW

**Comment by ronaldtse** [comment]
Thanks @catap , just wanted to remind to use our Ribose BSD 2-clause header as seen here for all code not inherited from NetPGP:

https://github.com/riboseinc/retrace/blob/master/common.c#L2-L23

**Comment by catap** [comment]
@ronaldtse I keep this email as unread to don't miss it ;)

and update of this task will be after this ;)

**Comment by dewyatt** [comment]
@catap KBX and G10 should not be separate formats, but a single option that uses `pubring.kbx` for pubkeys and `private-keys-v1.d` for seckeys. There is no such thing as `public-keys-v1.d`.

(Correct me if I'm wrong)

**Comment by catap** [comment]
@dewyatt sure, it will be. And I will fix it, when it will be done.

Right now it is just a first step to prevent huge-few weeks PR that difficult to merge in ;)

**Comment by dewyatt** [comment]
@catap Ok that sounds good, but let's get it building and passing tests first.

Also, in the future, can you use hyphen-separated filenames like in the dev guide? (`packet-parse.c`, etc) Not important right now though.

**Comment by catap** [comment]
@dewyatt surre, and I expect that it will be built.

Anyway if you check the log you can see that it has a problem on "autohell level" :(

File: src/tests/support.c [hunk 1]
```diff
@@ -330,16 +332,13 @@ setupPassphrasefd(int *pipefd)
...
 {
     int          res;
-    char         pubpath[1024];
-    char         secpath[1024];
-    char         homepath[1024];
-    rnp_params_t params;
+    char         pubpath[MAXPATHLEN];
+    char         secpath[MAXPATHLEN];
+    char         homepath[MAXPATHLEN];
+    rnp_params_t params = {};
```
Comment (dewyatt) [inline src/tests/support.c]: Please use `{0}` here to be compliant with the standard.

**Comment by catap** [comment]
@dewyatt I can't pass style hell again.

[Travis job](https://travis-ci.org/riboseinc/rnp/jobs/258836981) has one style error:
```
==== Style changes required ====
--- src/tests/generatekey.c	(before formatting)
+++ src/tests/generatekey.c	(after formatting)
@@ -243,9 +243,9 @@
 
     rnp_test_state_t *rstate = *state;
     const char *hashAlg[] = {"MD5", "SHA1", "SHA256", "SHA384", "SHA512", "SHA224", "SM3"};
-    const char *keystores[] = {RNP_KEYSTORE_GPG, RNP_KEYSTORE_KBX, RNP_KEYSTORE_G10};
-    rnp_t       rnp;
-    int         pipefd[2];
+    const char *      keystores[] = {RNP_KEYSTORE_GPG, RNP_KEYSTORE_KBX, RNP_KEYSTORE_G10};
+    rnp_t             rnp;
+    int               pipefd[2];
 
     for (int i = 0; i < sizeof(hashAlg) / sizeof(hashAlg[0]); i++) {
         for (int j = 0; j < sizeof(keystores) / sizeof(keystores[0]); j++) {
```

Ok, I fixed it by hand and when I try to commit the `pre-commit` hook said:
```
➜  rnp git:(librekey) ✗ git commit -a -m '.'
--- src/tests/generatekey.c	2017-07-29 14:26:05.000000000 +0400
+++ -	2017-07-29 14:26:05.000000000 +0400
@@ -243,9 +243,9 @@
 
     rnp_test_state_t *rstate = *state;
     const char *hashAlg[] = {"MD5", "SHA1", "SHA256", "SHA384", "SHA512", "SHA224", "SM3"};
-    const char *      keystores[] = {RNP_KEYSTORE_GPG, RNP_KEYSTORE_KBX, RNP_KEYSTORE_G10};
-    rnp_t             rnp;
-    int               pipefd[2];
+    const char *keystores[] = {RNP_KEYSTORE_GPG, RNP_KEYSTORE_KBX, RNP_KEYSTORE_G10};
+    rnp_t       rnp;
+    int         pipefd[2];
 
     for (int i = 0; i < sizeof(hashAlg) / sizeof(hashAlg[0]); i++) {
         for (int j = 0; j < sizeof(keystores) / sizeof(keystores[0]); j++) {

Formatting changes requested.
See /var/folders/rq/wqy42lq543z9rwrlj7vf1h4r0000gn/T/git-clang-format.XXXXXX.patch.qI1L9uAa

Apply now (y/n)? 
```

So, I can't commit changes that request Travis hook because local hooks required opposite :(

**Comment by dewyatt** [comment]
@catap It looks like the LLVM snapshot apt repos have been updated and travis-ci is likely using 4.0.1 now. Alpine is lagging behind so I'll have to put a bloated ubuntu container together I guess.

Anyways, for now, just ignore it.

**Comment by dewyatt** [comment]
> and yes, I moved keystore as librekey. I tried extract library for work with pgp, but it is a huge refactoring :(

Yeah most of the code base is very highly coupled and fully separating things would take months IMO.
*Some* things are nicely isolated, like `pgp_create_sig_t` is opaque and only fully defined in signature.c. It would probably be nice to do that for everything at some point, but it's a lot of work.

**Comment by dewyatt** [comment]
(In the interest of time, I bent the rules and merged.)

**Comment by dewyatt** [comment]
I'm going to go ahead and allow failures on the style check in CI as well, until I have time to fix this issue.

**Comment by ronaldtse** [comment]
Thanks @catap and @dewyatt !

**Comment by catap** [comment]
@dewyatt I agree with your estimation and I tried do some small refactoring but after few hours I realised how big it is.

**Pull request merged**